docs/how-to-create-a-machine.md

```diff
@@ -0,0 +1,167 @@
+# How to create a machine
+
+A `StateMachine a b` is a stateful process which receives inputs of type `a` and emits outputs of type `b`.
+
+We will represent a machine as such
+
+```
+a ┌────┐ b
+──┤    ├──
+  └────┘
+```
+
+as black boxes, where inputs are on the left and outputs are on the right.
+
+The `StateMachine a b` data type has four constructors which we can use to construct a machine:
+
+- `Basic`
+- `Compose`
+- `Parallel`
+- `Alternative`
+
+Let's start with the last three.
+
+## `Compose`
+
+```haskell
+Compose
+  :: StateMachine a b
+  -> StateMachine b c
+  -> StateMachine a c
+```
+
+allows us to sequentially compose two machines
+
+```
+a ┌────┐ b   b ┌────┐ c
+──┤    ├──   ──┤    ├──
+  └────┘       └────┘
+```
+
+to get a single machine
+
+```
+  ┌─────────────────┐
+a │ ┌╌╌╌╌┐ b ┌╌╌╌╌┐ │ c
+──┼╌┤    ├╌╌╌┤    ├╌┼──
+  │ └╌╌╌╌┘   └╌╌╌╌┘ │
+  └─────────────────┘
+```
```

Comment:
I certainly appreciate the effort that must have gone into drawing this intricate ASCII art. :laughing: 

Consider using [Mermaid](https://mermaid.js.org/syntax/stateDiagram.html) instead.

```mermaid
stateDiagram-v2
    direction LR
    [*] --> A
    state A {
        direction LR
        [*] --> B: a
        B --> C: b
        C --> [*]: c
    }
    A --> [*]
```

Reply:
I do love Mermaid, but for these diagram I kind of prefer to be able to control precisely how things are represented, especially for more complex diagrams like this one:

![Screenshot_20230131_095000](https://user-images.githubusercontent.com/2643972/215712893-cbd7eaef-6f5c-4c51-a18d-1c4f6cdbba28.png)

If in the future we will use Mermaid to generate similar diagrams automatically, then it would definitely make sense to use the same tool in the documentation